Services/HetrixtoolsService/AzineMailgunHetrixtoolsService.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Azine\MailgunWebhooksBundle\Services\HetrixtoolsService;
+
+/**
+ * AzineMailgunHetrixtoolsService
+ *
+ * This service is a wrapper for using Hetrixtools blacklist check functionality https://hetrixtools.com/.
+ */
+class AzineMailgunHetrixtoolsService
+{
+    /**
+     * @var string
+     */
+    private $apiKey;
+
+    /**
+     * @var string
+     */
+    private $blacklistIpCheckUrl;
+
+    /**
+     * AzineMailgunHetrixtoolsService constructor.
+     *
+     * @param string $apiKey
+     * @param string $url
+     */
+    public function __construct($apiKey, $url)
+    {
+        $this->apiKey = $apiKey;
+        $this->blacklistIpCheckUrl = $url;
+    }
+
+    /**
+     * @param string $ip
+     * @return HetrixtoolsServiceResponse $response
+     */
+    public function checkIpAddressInBlacklist($ip)
+    {
+        $url = $this->prepareBlacklistIpCheckUrl($ip);
+        $ch = curl_init($url);
+
+        curl_setopt($ch, CURLOPT_HEADER, true);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+
+        $server_output = curl_exec($ch);
+        $info = curl_getinfo($ch);
+        $header_size = curl_getinfo($ch, CURLINFO_HEADER_SIZE);
+        curl_close($ch);
+
+        $response = substr($server_output, $header_size);
+
+        if($info['http_code'] != 200){
+
+            return null;
```

Comment:
Shouldn't this throw an exception?! Currently any non 200-response is interpreted as "HetrixTools did not send a reply", but if the response indicates an error-state in the configuration or a problem with the api-key or URL, then we would never find out, because the only information we get is "no reponse" as opposed to "Error XY".

I think throwing an exception and handling it properly in the command would be better. What's your thoughts on this?